binding.gyp

```diff
@@ -1,68 +1,123 @@
+# This file inherits default targets for Node addons, see https://github.com/nodejs/node-gyp/blob/master/addon.gypi
 {
-    'includes': [ 'common.gypi' ],
-    'variables': {
-      # Includes we don't want warnings for.
+  # https://github.com/springmeyer/gyp/blob/master/test/make_global_settings/wrapper/wrapper.gyp
+  'make_global_settings': [
+    ['CXX', '<(module_root_dir)/mason_packages/.link/bin/clang++'],
+    ['CC', '<(module_root_dir)/mason_packages/.link/bin/clang'],
+    ['LINK', '<(module_root_dir)/mason_packages/.link/bin/clang++'],
+    ['AR', '<(module_root_dir)/mason_packages/.link/bin/llvm-ar'],
+    ['NM', '<(module_root_dir)/mason_packages/.link/bin/llvm-nm']
```

Comment:
Why do these compilation settings depend on mason? Folks should be able to compile with their system compiler, no? Can we set paths externally and keep the build system agnostic to mason?

Reply:
Yes, you can use an alternative compiler 👍 These are just defaults. See https://github.com/mapbox/node-cpp-skel/blob/master/README.md#customizing-the-compiler-toolchain for details.